src/main/java/com/cinema/controller/MovieSessionController.java

```diff
@@ -0,0 +1,45 @@
+package com.cinema.controller;
+
+import com.cinema.mapper.MovieSessionMapper;
+import com.cinema.model.MovieSession;
+import com.cinema.model.dto.MovieSessionRequestDto;
+import com.cinema.model.dto.MovieSessionResponseDto;
+import com.cinema.service.MovieSessionService;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper mapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper mapper) {
+        this.movieSessionService = movieSessionService;
+        this.mapper = mapper;
+    }
+
+    @PostMapping
+    public void add(@RequestBody MovieSessionRequestDto requestDto) {
+        MovieSession movieSession = mapper.mapFromReqDtoToEntity(requestDto);
+        movieSessionService.add(movieSession);
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getByParams(@RequestParam Long movieId,
+                                                     @RequestParam String date) {
```

Comment:
You can use LocalDate as method parametr. String is parsing to LocalDate on fly.

Reply:
agree